packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -929,8 +929,14 @@ export function performConcurrentWorkOnRoot(
           continue;
         }
 
-        // Check if something threw
-        if (exitStatus === RootErrored) {
+        // Check if something threw.
+        // We only need to retry during hydration or concurrent render.
```

Comment:
We shouldn't even do it for those because we should refactor the Root recovery to work the same way as Suspense recovery.